src/meta/meta_ckpt.c

```diff
@@ -1262,6 +1262,195 @@ __wt_meta_ckptlist_update_config(
+
+    return (0);
+}
+
+/*
+ * __get_blkmods --
+ *     Extracts the blkmod info from the metadata about a file+id into a bitmap in a WT_ITEM.
+ */
+static WT_INLINE int
+__get_blkmods(WT_SESSION_IMPL *session, const char *uri, const char *id, WT_ITEM *output_item)
+{
+    WT_CONFIG blkconf;
+    WT_CONFIG_ITEM backup_config_value, blocks, blocks_key, blocks_value, key;
+    WT_CURSOR *metadata_cursor;
+    WT_DECL_RET;
+    char *file_config;
+
+    WT_CLEAR(backup_config_value);
+    WT_CLEAR(blocks);
+    WT_CLEAR(blocks_key);
+    WT_CLEAR(blocks_value);
+    WT_CLEAR(key);
+    WT_CLEAR(*output_item);
+
+    metadata_cursor = NULL;
+    file_config = NULL;
+
+    WT_RET(__wt_metadata_cursor_open(session, NULL, &metadata_cursor));
+
+    metadata_cursor->set_key(metadata_cursor, uri);
+    WT_ERR(metadata_cursor->search(metadata_cursor));
+    WT_ERR(metadata_cursor->get_value(metadata_cursor, &file_config));
+    WT_ERR(
+      __wt_config_getones(session, file_config, "checkpoint_backup_info", &backup_config_value));
+
+    if (backup_config_value.len > 0) {
+        if (backup_config_value.type == WT_CONFIG_ITEM_STRUCT) {
+            __wt_config_subinit(session, &blkconf, &backup_config_value);
+
+            /* Loop through the incremental backup blocks data looking for the correct id */
+            while ((ret = __wt_config_next(&blkconf, &blocks_key, &blocks_value)) == 0) {
+                if (blocks_value.len > 0) {
+                    if (WT_STRING_MATCH(id, blocks_key.str, blocks_key.len)) {
+                        /* We've found the right blocks so read the bit pattern into output_item */
+                        ret = __wt_config_subgets(session, &blocks_value, "blocks", &blocks);
+                        if (ret == 0) {
+                            if (blocks.len > 0) {
+                                ret =
+                                  __wt_nhex_to_raw(session, blocks.str, blocks.len, output_item);
+                                break;
+                            }
+                        }
+                        WT_ERR_NOTFOUND_OK(ret, false);
+                    }
+                }
+            }
+        }
+    }
+
+    if (ret == WT_NOTFOUND)
+        ret = 0;
+
+err:
+    WT_TRET(metadata_cursor->close(metadata_cursor));
+
+    return (ret);
+}
+
+/*
+ * __check_incorrect_modified_bits --
+ *     This function takes as input two bitmaps (in WT_ITEMs), an original and a new. If any bit in
+ *     the original changed from 1 to 0, it's an error, and the new bitmap is NOT ok. Otherwise, the
+ *     new bitmap is ok.
+ */
+static WT_INLINE int
+__check_incorrect_modified_bits(WT_ITEM *original_bitmap, WT_ITEM *new_bitmap, bool *ok)
+{
+    size_t index;
+    uint8_t *new_ptr, *original_ptr;
+    uint8_t partial_result;
+
+    WT_NOT_READ(index, 0);
+    original_ptr = NULL;
+    new_ptr = NULL;
+    WT_NOT_READ(partial_result, 0);
```

Comment:
Is `WT_NOT_READ` needed?

Reply:
We need it because otherwise we get the following errors from the clang analyser:

```
[2024/03/17 19:54:29.018] /data/mci/8e9fbae01b34d2314f0ee58c6aa5e59e/wiredtiger/src/meta/meta_ckpt.c:1362:5: warning: Value stored to 'index' is never read [deadcode.DeadStores]
[2024/03/17 19:54:29.018]     index = 0;
[2024/03/17 19:54:29.018]     ^       ~
[2024/03/17 19:54:29.018] /data/mci/8e9fbae01b34d2314f0ee58c6aa5e59e/wiredtiger/src/meta/meta_ckpt.c:1365:5: warning: Value stored to 'partial_result' is never read [deadcode.DeadStores]
[2024/03/17 19:54:29.018]     partial_result = 0;
```

Reply:
I don't understand though, it is read [here](https://github.com/wiredtiger/wiredtiger/pull/9437/files#diff-3be0ed5707ef4b0f982eac862ff81534805bd6a7c363fa4241d8a09534e1cb20R1397).

Reply:
I also don't understand why the errors are generated in this case because, as you say, the variables are read. My current theory is that the compiler thinks there are some paths where the written values are never subsequently read, but that requires further investigation.